### DISCUSSION TITLE
Suricata dataset rule fails to load it's file

### DISCUSSION THREAD

**Discussion by Drongo-Px** [Security-Onion-Solutions/securityonion#7868] [Unsupported Versions]
I have created a dataset rule that runs against a list of ja3 base64 hash values but it keeps triggering errors, these errors indicate that the file provided in the dataset rule does not exist, but the file exists!!.
![ja3 fopen](https://user-images.githubusercontent.com/73431385/165599811-46f4197d-7437-4d3e-b5eb-785a68dc38e1.jpg)

Any idea why Suricata keeps triggering this error while testing
Thanks

**Comment by tomjones1977** [comment]
What are the file permissions on the ja3.dataset file?

**Reply by Assem-Murad** [reply]
@tomjones1977 I changed it to 777 to ensure that the permission isn't the root cause
![suriperm](https://user-images.githubusercontent.com/96170392/165714969-2e609b6d-4415-4ce8-9d39-6d5f8fc8f1c8.jpg)

**Reply by tomjones1977** [reply]
Is that file on the local filesystem or in the docker container?  Have you edited the container to mount that location?

**Reply by Assem-Murad** [reply]
> Is that file on the local filesystem or in the docker container? Have you edited the container to mount that location?

in the local filesystem

**Reply by tomjones1977** [reply]
By default the suricata container has access to the following locations on the host:

```
        "HostConfig": {
            "Binds": [
                "/opt/so/conf/suricata/suricata.yaml:/etc/suricata/suricata.yaml:ro",
                "/opt/so/conf/suricata/threshold.conf:/etc/suricata/threshold.conf:ro",
                "/opt/so/conf/suricata/rules:/etc/suricata/rules:ro",
                "/opt/so/log/suricata/:/var/log/suricata/:rw",
                "/nsm/suricata/:/nsm/:rw",
                "/nsm/suricata/extracted:/var/log/suricata//filestore:rw",
                "/opt/so/conf/suricata/bpf:/etc/suricata/bpf:ro"
```

The left hand side is the local file/folder, the right hand side is where it's mounted in the container filesystem.  Try copying your file to `/opt/so/conf/suricata/rules/ja3.dataset` and changing the file reference to `/etc/suricata/rules/ja3.dataset`

**Reply by Assem-Murad** [reply]
> By default the suricata container has access to the following locations on the host:
> 
> ```
>         "HostConfig": {
>             "Binds": [
>                 "/opt/so/conf/suricata/suricata.yaml:/etc/suricata/suricata.yaml:ro",
>                 "/opt/so/conf/suricata/threshold.conf:/etc/suricata/threshold.conf:ro",
>                 "/opt/so/conf/suricata/rules:/etc/suricata/rules:ro",
>                 "/opt/so/log/suricata/:/var/log/suricata/:rw",
>                 "/nsm/suricata/:/nsm/:rw",
>                 "/nsm/suricata/extracted:/var/log/suricata//filestore:rw",
>                 "/opt/so/conf/suricata/bpf:/etc/suricata/bpf:ro"
> ```
> 
> The left hand side is the local file/folder, the right hand side is where it's mounted in the container filesystem. Try copying your file to `/opt/so/conf/suricata/rules/ja3.dataset` and changing the file reference to `/etc/suricata/rules/ja3.dataset`

thanks for your fast reply 
I have already tested this file at this path /opt/so/conf/suricata/rules/ja3.dataset and still triggering the same error

**Reply by tomjones1977** [reply]
When you put the file in `/opt/so/conf/suricata/rules/ja3.dataset` what did you have in your rules file as the path?

**Reply by Assem-Murad** [reply]
> When you put the file in `/opt/so/conf/suricata/rules/ja3.dataset` what did you have in your rules file as the path?
Here is a screenshot that shows what you have mentioned regarding the path
![image](https://user-images.githubusercontent.com/96170392/165717342-451614c7-62f4-4f71-b8fb-66339c8882af.png)

**Reply by tomjones1977** [reply]
The path on the host and the path in the container are not the same.  You need to change the file reference to `/etc/suricata/rules/ja3.dataset`.

**Reply by Assem-Murad** [reply]
> The path on the host and the path in the container are not the same. You need to change the file reference to `/etc/suricata/rules/ja3.dataset`.

Thanks,i will test it right now and i will update you with the results

**Reply by Assem-Murad** [reply]
> The path on the host and the path in the container are not the same. You need to change the file reference to `/etc/suricata/rules/ja3.dataset`.
i changed the file path in the suricata rule and still triggering the same error 
![stillthesame](https://user-images.githubusercontent.com/96170392/165718889-abacc1d5-2d9c-444f-bcd5-9a15c76be806.jpg)

**Reply by tomjones1977** [reply]
Check that the file appears in the container.  `docker exec so-suricata ls -l /etc/suricata/rules`  Can you see the file if you run this command?

**Reply by Assem-Murad** [reply]
> Check that the file appears in the container. `docker exec so-suricata ls -l /etc/suricata/rules` Can you see the file if you run this command?
Yes i can see it 
![c](https://user-images.githubusercontent.com/96170392/165722529-72bf9ac8-c8d9-40dc-a359-3e4eec474bef.jpg)

**Reply by tomjones1977** [reply]
This appears to be an issue with how the `so-suricata-testrule` loads files into the temporary docker container.  Looking at the testrule script shows it uses a temporary container not the actual so-suricata container.  I can't see a straightforward way to include other files without modifying the testrule script.  You could try copying the ja3.dataset to `/tmp/nids-testing/rules`, but this is a bodge just to get the test to run.

**Reply by Assem-Murad** [reply]
> This appears to be an issue with how the `so-suricata-testrule` loads files into the temporary docker container. Looking at the testrule script shows it uses a temporary container not the actual so-suricata container. I can't see a straightforward way to include other files without modifying the testrule script. You could try copying the ja3.dataset to `/tmp/nids-testing/rules`, but this is a bodge just to get the test to run.

unfortunately still giving the same error even after modifying the path to the one you just mentioned.
![still](https://user-images.githubusercontent.com/96170392/165726568-d96642c2-0705-4482-9d1c-bd79c60d577b.jpg)

**Reply by tomjones1977** [reply]
That's not what I meant.  The path is the same from a suricata perspective, but the folder that is mounted is the /tmp one.
